### ISSUE TITLE
Viewer seems to only support 1 browser tab

### ISSUE THREAD

**Issue by Charence** [3drepo/3drepo.io#3992]
### Description

Nothing appears in the 3D viewer when more than 1 tab of 3D Repo is running in the same browser.

- This is happening in: ~prod/staging/~ both 
- This requires a hotfix to production: yes/no 

### Steps to replicate
- Open a model in 3drepo.io
- Duplicate tab

### Current Behaviour
- Only the first 3drepo.io tab has a visible 3D model
- Other 3drepo.io tabs with the same URL have an empty viewer

### Expected Behaviour
- All tabs should load the 3D model

### Pictures/ Gifs of the error
![Screenshot from 2023-02-20 13-17-48](https://user-images.githubusercontent.com/1438332/220121924-1566aad4-ca70-4612-af4b-9d32bf4551d2.png)
![Screenshot from 2023-02-20 13-17-51](https://user-images.githubusercontent.com/1438332/220121929-e3263629-730a-4289-bf29-d40da6937631.png)
![Screenshot from 2023-02-20 13-17-53](https://user-images.githubusercontent.com/1438332/220121932-e90bbf99-828b-46a5-bf77-b45fa8c23be5.png)
![Screenshot from 2023-02-20 13-18-59](https://user-images.githubusercontent.com/1438332/220121946-2801f204-dc96-463a-8558-b43403e6be95.png)

Same in FF:
![Screenshot from 2023-02-20 13-24-29](https://user-images.githubusercontent.com/1438332/220121969-217ac776-e100-4ba2-8605-cf18963e2ca5.png)

**Comment by sebjf** [comment]
This is almost certainly because emscripten is not surrendering its connection.

The solution should be for `clearIdbfs` to check for `onblocked` and `onupgradeneeded`, to detect when an open connection blocks the delete request and abort its attempt.

However, implementing this does not work: one tab successfully detects `onblocked`, but then the viewer stalls at 90% (perhaps the idbfs functionality in Unity is broken?) and subsequent tabs simply stall as initially reported (none of the callbacks are ever made by IndexedDb).

The solution will probably have to be to detect if another tab is open and not try to delete the db in that case, or, to enumerate the files in Unity (if possible) and delete them that way (the downside of that is it will still need to load the whole cache once before it deletes it all, which undermines the point for customers who are crashing on load).

@carmenfan do you have a preference? I am just seeing now if its possible to make Unity enumerate the cache.

**Comment by sebjf** [comment]
>  to detect if another tab is open

A thought, the solution doesn't need to detect if a tab is open - simply whether the clear operation has ever run or could be running.

We could try checking for and setting a key on local storage before starting the delete. If the key is found the clear operation is skipped. This should work so long as localStorage operations across tabs are atomic.

This would be much simpler and have less potential for bugs than trying to get a live tab count.

**Comment by carmenfan** [comment]
> We could try checking for and setting a key on local storage before starting the delete. If the key is found the clear operation is skipped. This should work so long as localStorage operations across tabs are atomic.
> This would be much simpler and have less potential for bugs than trying to get a live tab count.

Agreed, I think this is teh better approach

**Comment by carmenfan** [comment]
Merged into #3887
